pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/NativeMutableTextIndex.java

```diff
@@ -114,8 +121,8 @@ public void close()
   private List<String> analyze(String document) {
     List<String> tokens = new ArrayList<>();
     try (TokenStream tokenStream = _analyzer.tokenStream(_column, document)) {
-      tokenStream.reset();
       CharTermAttribute attribute = tokenStream.getAttribute(CharTermAttribute.class);
+      tokenStream.reset();
```

Comment:
this doesn't change behavior, but follows the intended flow from the [docs](https://lucene.apache.org/core/4_7_0/core/org/apache/lucene/analysis/TokenStream.html)